vaadin-date-picker-flow-parent/vaadin-date-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datepicker/DatePickerLocaleIT.java

```diff
@@ -212,4 +239,30 @@ private void assertNoWarnings() {
                     logEntry));
         }
     }
+
+    private void submitValue(String value) {
+        TestBenchElement input = picker.findElement(By.tagName("input"));
+        while (!input.getAttribute("value").isEmpty()) {
+            input.sendKeys(Keys.BACK_SPACE);
+        }
+        input.sendKeys(value);
+        input.sendKeys(Keys.ENTER);
+        getCommandExecutor().waitForVaadin();
+    }
+
+    private void assertInputOutputPair(String input, String expectedOutput) {
+        submitValue(input);
+        Assert.assertEquals(expectedOutput, $("span").id("output").getText());
```

Comment:
Using an output span is not really necessary for these tests, checking the parsed client value would be enough. The value can be accessed with `picker.getDate()`, see the existing tests above.

Reply:
Addressed in the latest commits